filip/clients/ngsi_v2/cb.py

```diff
@@ -880,18 +881,39 @@ def get_subscription_list(self,
             self.log_error(err=err, msg=msg)
             raise
 
-    def post_subscription(self, subscription: Subscription) -> str:
+    def post_subscription(self, subscription: Subscription,
+                          update: Optional[bool] = False) -> str:
         """
         Creates a new subscription. The subscription is represented by a
         Subscription object defined in filip.cb.models.
 
-        Args:
-            subscription:
+        If the subscription already exists, the adding is prevented and the id
+        of the existing subscription is returned.
 
+        Args:
+            subscription: Subscription
+            update: True - If the subscription already exists, update it
+                    False- If the subscription already exists, throw warning
         Returns:
-            object: 
+            str: Id of the (created) subscription
 
         """
+        existing_subscriptions = self.get_subscription_list()
+
+        sub_hash = str(subscription.dict(include={'subject', 'notification'}))
+        for ex_sub in existing_subscriptions:
+            if sub_hash ==\
+                    str(ex_sub.dict(include={'subject', 'notification'})):
+                self.logger.info("Subscription already exists")
+                if update:
```

Comment:
This only checks for existence of update meaning that any variable but "0, False, None" will trigger this

Reply:
How would you recommend checking the condition? As update is of type bool and has the default value False, this should be the correct logic. The user can technically enter a string as update parameter, but he will get a type notification. 

I can change it to update==True, to only allow True as correct value, but I think that derives from the standard code style guidelines. What do you think?

Reply:
This does fit now :)